pinot-query-runtime/src/main/java/org/apache/pinot/query/service/dispatch/QueryDispatcher.java

```diff
@@ -245,6 +323,14 @@ private DispatchClient getOrCreateDispatchClient(QueryServerInstance queryServer
     return _dispatchClientMap.computeIfAbsent(key, k -> new DispatchClient(hostname, port));
   }
 
+  private TimeSeriesDispatchClient getOrCreateTimeSeriesDispatchClient(
+      TimeSeriesQueryServerInstance queryServerInstance) {
+    String hostname = queryServerInstance.getHostname();
+    int port = queryServerInstance.getQueryServicePort();
+    String key = String.format("%s_%d", hostname, port);
```

Comment:
Is this key generation similar to sql query dispatcher? I am wondering if hostname changes for pod specially in kubernetes cluster what would be the side effect?

Reply:
This is the same as the Multistage Engine so all related constraints will apply. In case of a hostname change, that will involve a node restart which will anyways lead to a query failure. Right now our Multistage queries aren't able to still run if any of the involved server dies or gets restarted midway.